### ISSUE TITLE
Mojave fresh install of OS & Lulu, incomplete blocking

### ISSUE THREAD

**Issue by zaphodd** [objective-see/LuLu#107]
Worringly, on a Mojave 10.14.1 fresh install, minutes later I downloaded the latest Lulu 1.1.2 app and my backed-up rule.plist applied (from seemingly correctly working Yosemite and High Sierra machines) - and, as I want to limit system apps, Mojave App Store still connects perfectly, even though the various rules are blocking them.   This is the second time i've installed Mojave this week and found the same ting.
As a test, I allowed only MDNSresponder, the 2 normal Lulu instances and my browser, yet again, Mac App Store was able to connect - without prompt.
Then I installed my custom HOSTS file, which blocked all Apple & related domains (it's over 2.5Mb) Still Mojave's App Store connected without prompt, I installed Privoxy, (a local proxy) and Mac App Store then refused to connect, until I allowed Privoxy in Lulu.

I have suspicions that Mojave's App Store is using some other process and bypassing LuLu, as these same experiments absolutely block App Store etc. connecting on other OS's.

I know this is likely a similar description to a couple of others, but I have no way of knowing if it's the same problem, despite similar sounding, apparent, results.

**Comment by zaphodd** [comment]
quick update:   same computer, uninstalled LuLu 1.1.2.  and installed. LuLu 0.9.5
and strangely, LuLu 0.9.5 blocks  everything as expected and now gives prompts when things like Mojave App Store tries to connect, where, with LuLu 1.1.2 didn't ask or block.

**Comment by zaphodd** [comment]
initially, yes. and I'm no stranger to CLI, firewalls, UI or Linux or Unix type systems, so as i said, on several seperate occasions, each on freshly installed Mojave,  i installed the latest Lulu, then after rebooting, unticked the 'allow Apple' and 'already installed applications' boxes, unticked apps i didn't want comunicating, but they STILL communicated.
 As a test, i only allowed MDNSresponder and my browser - but ALL connections were not blocked in spite of the Lulu UI saying otherwise. I also clicked 'apply to all apps, whatever their location'
This failed also  and even after a reboot to flush any possible caches, LULU continued to ignore any rules i set :(       a saved working ruleset (from Lulu 0.95) was tried several times to no avail.
The Lulu install seemed to go as expected, but it does not block in my tests, it only blocks (or asks) when i run 3rd party apps like Firefox etc. All Apple services, daemons still comunicate outbound.

Lulu 0.95 works exactly as expected.  Lulu 0.95 correctly blocks things like SystemUIServer etc and notifies correctly, Lulu 0.95 blocks all internal apps. 

This happens on 3 Macs,  each on seperately and freshly installed OS, each running Mojave.
this has happens every time i come to try recent Lulu releases, three different computers.
i have to then  uninstall and roll back to Lulu 0.95 which blocks any and all services listed in the UI i choose.  I have tried both with SIP enabled and disabled.

to recap, I install (recent) Lulu normally, on fresh Mojave installs, disallow Apple and 3rd party apps, but it does not block any system apps other than the apps i install like Firefox etc.

The only way Lulu blocks anything in it's UI i choose is to use Lulu 0.95 :(
this is on 3 seperate Macs :(

**Comment by air-10** [comment]
same here! Lulu seems not stopping anything. All the settings intact. 10.14.4

**Comment by zaphodd** [comment]
It seems they either don't care, don't believe us or can't be bothered investigating our bug reports :(

**Comment by air-10** [comment]
Yes. I guess it’t time for a percolator. Oops, sorry. Time to get back to little snitch. It works good and do not pass anything.

**Comment by zaphodd** [comment]
Ye, i guess so :(   chai latte time here :)

**Comment by alvarnell** [comment]
"They" is only one developer and he now has a pretty much full-time job with at least one commercial firm, along with hosting a couple of conferences since last working on this, so it isn't surprising to see the project be on a back burner for now and I don't have any clues when or if he will have time to return to it.

I returned to LS many many months ago, for that reason.

**Comment by stepini** [comment]
So the project is closed... bad news

**Comment by f1ght3r** [comment]
the last version had fixed all the bugs i encoutered lulu v1.2.0 i think it was beta when i installed it but had no problems since

**Comment by zaphodd** [comment]
It's the OS that seems to be the issue. Lulu is pretty much fine. Lulu can't block stuff it can't see.
running my own DNS and blocking Apple IPs still much Apple stuff wriggles its way through to the internet.
I don't trust Apple any further than i would trust any multinational or big business or organisation.
If you can't trust your (the) OS then you can never truly be 100% safe or private.

~Almost like Apple rootkitted their OS - but they didn't, I'm sure they had their reasons.
Same with windows,  you can add HOSTS files and all kinds of stuff, still most sneaky software
 finds it's way to the internet.
 
Seems like OS X - anything past Mojave is pretty much an open book, I will be heading back to Linux or BSD after 10 break.
Pity.   i like Early OS X, it even looked nice too. i will use it for my music production while disconnected from the 'net.
